subprojects/distributions/src/integTest/groovy/org/gradle/SrcDistributionIntegrationSpec.groovy

```diff
@@ -50,6 +51,7 @@ class SrcDistributionIntegrationSpec extends DistributionIntegrationSpec {
             inDirectory(contentsDir)
             usingExecutable('gradlew')
             withTasks('binZip')
+            withEnvironmentVars(repoMirrorUrlsEnvironment())
```

Comment:
Maybe I'm wrong, but in my understanding, don't all processed forked by Gradle inherit all environment variables? Why we need to pass these env variables manually?

Reply:
Good point. Why did this test fail: https://builds.gradle.org/viewLog.html?buildId=14871612&buildTypeId=Gradle_Check_BuildDistributions
I thought we are already mirroring the Kotlin DSL repos.

Reply:
Because kotlin-dsl plugin would add such a repository to the project: https://github.com/gradle/kotlin-dsl/blob/3b07ab2383ebe06f0d05cf0e6c70d9993713ae43/subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/RepositoryHandlerExtensions.kt#L106

@eskatos How can we get rid of this "default" repository?

Reply:
Hmm... Don't we change the URL of this repository in the `mirrors` script?

Reply:
Well, this repository was added extraly by kotlin-dsl plugin, not us. We don't change anything in `mirror` script, we just use the mirror url if it exists in environment variables.

Reply:
How about we use something like the `MirrorPlugin` we use in the integration tests in the Gradle build itself? That can be part of a shared script plugin I suppose.

Reply:
Good idea. Let's make it in the recent https://github.com/gradle/gradle/pull/6449

Reply:
Done in #6449. And I think we can revert this PR because this is unnecessary.

Reply:
We should.